### PR TITLE
replaced elastic search java client for es query  with http client  to minimize dependencies of plugin

### PR DESCRIPTION
**Pull request by prudhvije**

### CODE REVIEW

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java [hunk 1]
```diff
@@ -48,23 +50,36 @@
...
+    private static final String SOURCE = "_source";
+    private static final String FILE_CONTENT = "fileContent";
+    private static final String IMPORT = "import ";
+    private static final String SORT_ORDER = "desc";
+    private static final String APPLICATION_JSON = "application/json";
+    private static final String FILE = "file";
+    private static final String TOKENS = "tokens";
+    private static final String CUSTOM_TOKENS_IMPORT_NAME = "custom.tokens.importName";
+    private static final char CH = '.';
+    private static final String HTTP_172_16_12_201_9201_SOURCEFILE_SEARCH = "http://172.16.12.201:9201/sourcefile/_search";
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java]: introduce a proper name

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java [hunk 2]
```diff
@@ -104,54 +117,74 @@ private DefaultMutableTreeNode getNodes(String projectName, List<CodeInfo> codeI
...
 
     public void runAction(final AnActionEvent e) throws IOException {
-        Editor projectEditor = DataKeys.EDITOR.getData(e.getDataContext());
-
-        if (projectEditor == null) {
-            //no editor currently available
-            return;
-        }
+        final Editor projectEditor = getEditor(e);
+        if (projectEditor == null) return;
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java]: no return

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java [hunk 3]
```diff
@@ -104,54 +117,74 @@ private DefaultMutableTreeNode getNodes(String projectName, List<CodeInfo> codeI
...
+        model.reload(root);
+        jTree.addTreeSelectionListener(getTreeSelectionListener(root));
+    }
+
+    private Editor getEditor(AnActionEvent e) {
+        final Editor projectEditor = DataKeys.EDITOR.getData(e.getDataContext());
+
+        if (projectEditor == null) {
+            //no editor currently available
+            return null;
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java]: same here

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java [hunk 4]
```diff
@@ -310,42 +300,130 @@ private BoolQueryBuilder setMustToken(BoolQueryBuilder boolQueryBuilder, String
...
+
+        List<ESQuery.Must> mustList = new ArrayList<ESQuery.Must>();
+
+        ESQuery.Must must;
+        ESQuery.Term term;
+
+        for (String nextImport : importsInLinesSet) {
+            must = new ESQuery.Must();
+            bool.setMust(mustList);
+            bool.setMust_Not(new ArrayList<ESQuery.Must>());
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java]: setMustNot

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java [hunk 5]
```diff
@@ -310,42 +300,130 @@ private BoolQueryBuilder setMustToken(BoolQueryBuilder boolQueryBuilder, String
...
+
 
-    private String getLocalPackages(Editor editor, Project project) {
-        PsiPackage pack = JavaPsiFacade.getInstance(project).findPackage("");
-        PsiPackage[] subPackages = pack.getSubPackages();
+    private static String getESResultJson(String esQueryJson, String url) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            HttpClient httpClient = new DefaultHttpClient();
+            HttpPost postRequest = new HttpPost(
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/BetterDocsAction.java]: unnecessary linebreak

**Comment by prudhvije** [comment]
new PR #42 with specified changes.

**Pull request closed without merging**